dialogue-clients/src/main/java/com/palantir/dialogue/clients/ServicesConfigBlockWithResolvedHosts.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.dialogue.clients;
+
+import com.google.common.collect.ImmutableSetMultimap;
+import com.palantir.conjure.java.api.config.service.ServicesConfigBlock;
+import com.palantir.logsafe.DoNotLog;
+import com.palantir.logsafe.logger.SafeLogger;
+import com.palantir.logsafe.logger.SafeLoggerFactory;
+import java.net.InetAddress;
+import java.util.Objects;
+
+@DoNotLog
+final class ServicesConfigBlockWithResolvedHosts {
```

Comment:
It might be a bit safer to use immutables to generate these sorts of types, since it'll include all fields in equals/hashCode